Python/compile.c

```diff
@@ -4474,10 +4489,17 @@ compiler_with(struct compiler *c, stmt_ty s, int pos)
     else if (!compiler_with(c, s, pos))
             return 0;
 
-    /* End of try block; start the finally block */
+    /* End of try block; call __exit__(None, None, None) */
+    PyObject *args = PyTuple_Pack(3, Py_None, Py_None, Py_None);
+    if (args == NULL) {
+        return 0;
+    }
+    ADDOP_N(c, LOAD_CONST, args, consts);
     ADDOP(c, POP_BLOCK);
-    ADDOP(c, BEGIN_FINALLY);
     compiler_pop_fblock(c, WITH, block);
+    ADDOP_I(c, CALL_FUNCTION_EX, 0);
+    ADDOP(c, POP_TOP);
+    ADDOP_JREL(c, JUMP_FORWARD, end);
 
     compiler_use_next_block(c, finally);
     if (!compiler_push_fblock(c, FINALLY_END, finally, NULL))
```

Comment:
Can you update the comment three lines below this? Currently it says:
```
     /* Finally block starts; context.__exit__ is on the stack under
       the exception or return information.
````

I would expect "Except block" instead of "Finally block", also "return information" doesn't apply anymore.

Reply:
Same comment for `compiler_async_with`, btw.